my-bq-app/src/pages/Login.jsx

```diff
@@ -0,0 +1,76 @@
+import React, {useState} from "react";
+import firebaseApp from "../firebase/credentials";
+import { getAuth, createUserWithEmailAndPassword, signInWithEmailAndPassword } from "firebase/auth";
+import { getFirestore, doc, setDoc } from "firebase/firestore";
+import BasicTextFields from '../components/BasicTextFields.js';
+import InputAdornments from "../components/InputAdornments.js";
+import SelectVariants from "../components/SelectVariants";
+
+const auth = getAuth(firebaseApp);
+const firestore = getFirestore(firebaseApp);
+
+function Login () {
+    const [isRegistered, setIsRegistered] = useState(false);
+
+    async function userSignUp(email, password, role) {
+      const userInfo = await createUserWithEmailAndPassword(
```

Comment:
Would it be better to move the firebase functions to their own file? To separate responsabilities, also for testing purposes. If you do decide to move them, you could put them in a folder named lib like we've done in other projects :)

Reply:
I already have a firebase folder, I think I could put it there, thank you :D